Update left turn approaches

avetraffic opened the pull request:
-Fix ped delay so that ped recall is not displayed if no red-to-red cycles are found
-Fix signal data check report to not break if there is an error in the backend and no view models are created.
-Fix left turn gap to use correct signal and approach versions if checking older data.
-Remove ability to check and uncheck checkboxes in the signalDetail display.
-Fix area bug*

*The FilterCheck uses indexOf to match areas to pins. The issue was that "1" also exists in "11", "12", "21". Since the list of areas is passed as a single string, to fix it, I've changed it to require commas on either side. So it now looks for ",1," instead. It's kinda ugly, but I didn't know how much performance we'd lose by parsing it into something more legible, and I didn't want to spend too much time on this because I think we're planning on replacing this in 5.0 or a version soon after. 

The signal repo change I mentioned is the Signal  GetVersionOfSignalByDate method. 
I've updated it to include the AddSignalAndDetectorLists needed to find the left turns.
I don't think adding it should create any issues, and I've tested all the metrics to make sure they still work after changing it. We can extract it into it's own method though if you prefer.

avetraffic committed:
Ped recall fix for when no red to red cycles are found

avetraffic committed:
Fix left turn run checks error if no signal data check report view models exist

avetraffic committed:
Fix Left Turn Gap Analysis to get correct signal version

avetraffic committed:
Remove ability to check and uncheck checkboxes in signal detail

avetraffic committed:
Add greater than sign to signal data check report

avetraffic committed:
Fix Area bug

dloweAvenueConsultants reviewed (changes requested):
1. If I am not mistaken you change to the Signal.cs file will never evaluate as empty or null.
2. Why are you using the view bag in the LeftTurnGapController. If possible we should add it to the view model.

avetraffic commented:
1. ~~I ran a non-signal number through it and it returned null for me.~~ I just noticed you mentioned the Signal.cs instead of the Signal repo, my bad. Good catch. I removed the line. Is there a better way to do this you think? I checked the metric filter too just to be sure but it looks like someone's already used a comma as a delimiter there so it should be fine.
2. I've removed the viewbag and just changed it from Model.first to Model.firstOrDefault. That should prevent the run checks error.

avetraffic committed:
remove unnecessary line

avetraffic committed:
Remove items from viewbag, change first -> firstOrDefault on signal data check

avetraffic committed:
fix approach volume chart

avetraffic committed:
update about page

avetraffic commented:
Additional changes:
-Fix error in Approach Volume Chart
-Update about page

dloweAvenueConsultants reviewed (approved):
Looks Good!
